File: app/src/main/java/com/tradepilot/data/model/Trade.kt
package com.tradepilot.data.model

enum class TradeDirection {
    BUY, SELL
}

enum class TradeResultStatus {
    WIN, LOSS
}

data class Trade(
    val id: String,
    val assetSymbol: String,
    val direction: TradeDirection,
    val amount: Double,
    val entryPrice: Double,
    val exitPrice: Double,
    val pnl: Double,
    val status: TradeResultStatus,
    val timestamp: String,
    val durationMinutes: Int = 5
)

File: backend/src/models/Trade.ts
export type TradeDirection = 'BUY' | 'SELL';
export type TradeResult = 'WIN' | 'LOSS';

export interface Trade {
  id: string;
  session_id: string;
  user_id: number;
  asset: string;
  direction: TradeDirection;
  amount: number;
  entry_price: number;
  exit_price: number;
  pnl: number;
  result: TradeResult;
  strategy: string;
  created_at: Date;
}
